packages/fastify-renderer/test/renderers/ReactRenderer.spec.ts

```diff
@@ -45,6 +48,37 @@ describe('ReactRenderer', () => {
     test.skip('should throw on rendering failure', async () => {
       return
     })
+
+    test('should call postRenderHooks after dom render', async () => {
+      const renderer = newReactRenderer()
+      const callOrder: string[] = []
+      renderer.plugin.hooks = [
+        {
+          heads: () => {
+            callOrder.push('heads')
+            return 'heads'
+          },
+          postRenderHeads: () => {
+            callOrder.push('postRenderHeads')
+            return 'postRenderHeads'
+          },
+        },
+      ]
+
+      renderer['renderSynchronousTemplate'](
```

Comment:
This is not ideal as we're testing a private method but calling the public `render` method involves setting up a bunch of `vite` related things. An option is to make this public, though I'm not sure if it should be